workers/transactor-api/src/index.ts

```diff
@@ -0,0 +1,138 @@
+//
+// Copyright © 2024 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import { gunzip } from 'zlib'
+import { promisify } from 'util'
+import {
+  Hierarchy,
+  type MeasureContext,
+  ModelDb,
+  type Class,
+  type Doc,
+  type DocumentQuery,
+  type FindOptions,
+  type FindResult,
+  type Ref,
+  type Tx,
+  type TxResult
+} from '@hcengineering/core'
+
+interface TransactorRawApi {
+  findAll: (_class: Ref<Class<Doc>>, query?: DocumentQuery<Doc>, options?: FindOptions<Doc>) => Promise<FindResult<Doc>>
+
+  tx: (tx: Tx) => Promise<TxResult>
+
+  getModel: () => Promise<Buffer>
+}
+
+export interface TransactorClient {
+  findAll: (_class: Ref<Class<Doc>>, query?: DocumentQuery<Doc>, options?: FindOptions<Doc>) => Promise<FindResult<Doc>>
+
+  tx: (tx: Tx) => Promise<TxResult>
+
+  getModel: () => Promise<ModelDb>
+}
+
+export interface TransactorService {
+  openRpc: (rawToken: string, workspaceId: string) => Promise<TransactorRawApi>
+}
+
+export async function unpackModel (compressed: Buffer): Promise<Tx[]> {
+  const ungzipAsync = promisify(gunzip)
+  const buffer = await ungzipAsync(new Uint8Array(compressed))
+  const decoder = new TextDecoder()
+  const jsonString = decoder.decode(buffer)
+  const model = JSON.parse(jsonString) as Tx[]
+  return model
+}
+
+export class TransactorRpcClient implements TransactorClient {
+  private transactorRpcStub: TransactorRawApi | undefined
+  private model: ModelDb | undefined
+
+  constructor (
+    private readonly ctx: MeasureContext,
```

Comment:
for end users ctx should be not require in the constructor, either create it when it needed or make it optional